src/ledger/insert_shred.zig

```diff
@@ -149,11 +152,15 @@ pub const ShredInserter = struct {
         const allocator = self.allocator;
         var reed_solomon_cache = try ReedSolomonCache.init(allocator);
         defer reed_solomon_cache.deinit();
+        // QUESTION: In scenario where shreds.len > 0 and not same as is_repaired.len
```

Comment:
`shreds.len > 0 and not same as is_repaired.len` this is illegal behavior. Maybe it is worth turning `debug` assert into real check with `panic`

Reply:
It's expected that the caller makes them the same length. If not, that's a bug, and you should not be surprised to see problems.

I think an assert here is fine since it's the kind of bug that would probably show up pretty quickly in testing. Also we typically use ReleaseSafe, so it is effectively a panic. If we really expect this to be a malformed input under some scenarios, I would have it return an error, not panic.